### DISCUSSION TITLE
Repeater: array of strings

### DISCUSSION THREAD

**Discussion by DSpeichert** [filamentphp/filament#3482] [Help]
Hi!

I am struggling to elegantly wrap a model attribute that has an `array` cast and contains simple strings:
```php
// model
protected $casts = [
    'ns'     => 'array',
];

// example
$model->ns = ['ns1.example.com', 'ns2.example.com'];
```

The "Repeater" functionality provides what I need - the ability to add arbitrary number of elements (within bounds). However, it does not seem to like the idea of the element being a flat string, schema requires a key.

Here's the code:

```php
Forms\Components\Repeater::make('ns')    // <- real key on the model
    ->label('DNS Servers')
    ->schema([
        Forms\Components\TextInput::make('somefakekey')    // <- not a real key, should be bare/inner
            ->label('DNS server hostname')
            ->required(),
    ])
    ->defaultItems(2)
    ->createItemButtonLabel('Add DNS server')
    ->minItems(2)
    ->maxItems(10)
    ->disableItemMovement()
```

I tried to cheat it a bit like this:

```php
    ->afterStateHydrated(fn ($component, $set, $state) => $set($component, collect($state)
        ->map(fn ($item) => ['somefakekey' => $item])
        ->toArray()
    ))
    ->dehydrateStateUsing(fn ($state) => collect($state)->pluck('somefakekey')->toArray()),
```

This approach works for displaying the form (`afterStateHydrated`) works, but apparently `dehydrateStateUsing` does not work on `Repeater`. It works on the actual fields of each repeated "element", but that's not where I can drop the fake key.

Is there any way to achieve this without a completely custom field?

**Comment by danharrin** [comment]
There is a way, but instead how about - https://filamentphp.com/plugins/simple-repeater
